File: src/python/repeat_the_character_in_string.py
"""
Code challenge: Repeat the characters

Create a Python function that accepts a string.
The function should return a string, with each character in the original string doubled.
If you send the function “now” as a parameter, it should return “nnooww,” 
and if you send “123a!”, it should return “112233aa!!”.
"""
def repeat_character(string_value):

    if not isinstance(string_value, str):
        raise ValueError("The argument value must be a string.")

    return ''.join([char*2 for char in string_value])


if __name__ == '__main__':

    test_value = 'now'
    result = repeat_character(test_value)
    print(f"Original: {test_value} | Doubled: {result}")

File: src/javascript/repeat_the_character_in_string.js
/**
 * Code challenge: Repeat the characters
 *
 * Create a function that accepts a string.
 * The function should return a string, with each character in the original string doubled.
 * If you send the function “now” as a parameter, it should return “nnooww,”
 * and if you send “123a!”, it should return “112233aa!!”.
 */

function repeatCharacters(stringValue) {
  const result = stringValue
    .split("")
    .map((c) => c + c)
    .join("");
  return result;
}

function main() {
  const testValue = "Found sounds";
  const result = repeatCharacters(testValue);
  console.log(`Original: ${testValue} | Doubled: ${result}`);
}

main();
